### ISSUE TITLE
Java bindings trouble

### ISSUE THREAD

**Issue by openbabel-bot** [openbabel/openbabel#1102]
There are serious problems with the Java bindings. It took me quite a while to get them working in both Linux and Mac OS. In Windows 7 x64 I had no problems. I just needed openbabel_java.dll from the OpenBabelGUI package to be in java.library.path.

Mac OS: I built the Java bindings using the instructions at [1] (including using Eigen2). However, it turns out that it is not sufficient to just place the resulting libopenbabel_java.so file in java.library.path. It needs to be renamed to libopenbabel_java.jnilib and the (advanced) instructions at [2] need to be followed because apparently there is some setup/compile glitch regarding where the compiled .so file will think its required library files are located.

Linux: Using "ld" I could see that the glitch from above is also present in Linux. I did not find a tool similar to "install_name_tool" so instead I found out that I needed to follow the instructions in the Java README [3]. There is a typo (openbabel_java.cpp needs to be openbabel-java.cpp) and there will be complaining about a "babelconfig.h" file which I copied from the build directory created using CMake as described in [4]. In the end, the resulting libopenbabel_java.so file needs to be in java.library.path and I needed to do the LD_PRELOAD trick.

To sum up, I think that there is room for improvement :) Thanks for your work so far!

[1] http://openbabel.org/docs/2.3.0/Installation/install.html#compile-bindings
[2] http://forums.openbabel.org/help-segmentation-fault-with-Pybel-on-Mac-OS-x-td3327036.html
[3] openbabel-2.3.0/scripts/java/README in openbabel-2.3.0.tar.gz
[4] http://openbabel.org/docs/2.3.0/Installation/install.html#basic-build-procedure

Reported by: mkjensen
